pkg/templates/encryption.go

```diff
@@ -50,30 +50,53 @@ func (t *TemplateResolver) protect(value string) (string, error) {
 // decrypt will decrypt a string that was encrypted using the protect method. An error is returned if the base64 or
 // the AES key is invalid.
 func (t *TemplateResolver) decrypt(value string) (string, error) {
+	// This is already validated in the NewResolver method, but is checked again in case that method was bypassed
+	// to avoid a panic.
+	if len(t.config.InitializationVector) != IVSize {
+		return "", ErrInvalidIV
+	}
+
 	decodedValue, err := base64.StdEncoding.DecodeString(value)
 	if err != nil {
 		return "", fmt.Errorf("%s: %w: %v", value, ErrInvalidB64OfEncrypted, err)
 	}
 
-	block, err := aes.NewCipher(t.config.AESKey)
-	if err != nil {
-		return "", fmt.Errorf("%w: %v", ErrInvalidAESKey, err)
-	}
+	var decryptionErr error
+	var decryptedValue []byte
 
-	// This is already validated in the NewResolver method, but is checked again in case that method was bypassed
-	// to avoid a panic.
-	if len(t.config.InitializationVector) != IVSize {
-		return "", ErrInvalidIV
+	var aesKeys [][]byte
+	if t.config.AESKeyFallback == nil {
+		aesKeys = [][]byte{t.config.AESKey}
+	} else {
+		aesKeys = [][]byte{t.config.AESKey, t.config.AESKeyFallback}
 	}
 
-	blockMode := cipher.NewCBCDecrypter(block, t.config.InitializationVector)
+	for _, aesKey := range aesKeys {
```

Comment:
Should this have a `break` on a successful decryption? It seems like it always goes on to try the fallback key... but I would expect it to return the result at the first point where it successfully decrypts.

Reply:
Thanks for catching this! I meant to add it but forgot to submit it.